lib/matplotlib/widgets.py

```diff
@@ -1237,35 +1250,108 @@ def _get_data(self, event):
         ydata = min(y1, ydata)
         return xdata, ydata
 
+    def _clean_event(self, event):
+        """Clean up an event
+
+        Use prev event if there is no xdata
+        Limit the xdata and ydata to the axes limits
+        Set the prev event
+        """
+        if event.xdata is None:
+            event = self._prev_event
+        else:
+            event = copy.copy(event)
+        event.xdata, event.ydata = self._get_data(event)
+
+        self._prev_event = event
+        return event
+
     def press(self, event):
-        """Button press handler"""
+        """Button press handler and validator"""
         if not self.ignore(event):
-            self.eventpress = copy.copy(event)
-            self.eventpress.xdata, self.eventpress.ydata = (
-                self._get_data(event))
+            event = self._clean_event(event)
+            self.eventpress = event
+            self._prev_event = event
+            key = event.key or ''
+            key = key.replace('control', 'ctrl')
```

Comment:
I don't really see this translating as a good thing to do. While the fact that we do "control" when the button is by itself and "ctrl" when it is in conjunction with something else isn't great, it is what people would see if they run the simple key-press reporting examples. Now, you are making it such that users need to target a diferent name than they expect (and it isn't even documented). I think we should remove the unneeded complexity and use `control` up above for the default dictionary.